### ISSUE TITLE
Change bayesian_optimizer.py

### ISSUE THREAD

**Issue by louisccc** [Sujit-O/pykg2vec#15]
In Bayesian hyperparameter tuner,  instead of using training loss, we should use the validation set's mean rank as a return value in the objective function.

**Comment by Sujit-O** [comment]
I fixed this issue. The Bayesian optimizer now gets the (mean_head_rank+mean_tail_rank)/2 as an accuracy measure from the validation set.
